ProjectName/src/screens/PaymentDetails/PaymentDetailsForm/components/customAmountInput/CustomAmountInput.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {FC} from 'react';
+import {Controller} from 'react-hook-form';
+import {View, Text, TextInput, Image} from 'react-native';
+
+import {DECIMAL_REGEX} from './custom-amount-input.constants';
+import {styles} from './custom-amount-input.styles';
+import {CustomMessageInputTypes} from '../customMessageInput/custom-message-input.types';
+
+const CustomAmountInput: FC<CustomMessageInputTypes> = ({
```

Comment:
I like how you approach creating separate `Input` components for different purposes. It's really smart and gives us flexibility.
The only thing I would advice to not pin such components to a single screen, but move them to a some general components folder. This way we could easily reuse them.